test/timeout_test.go

```diff
@@ -163,6 +163,67 @@ func TestPipelineRunTimeout(t *testing.T) {
 	}
 }
 
+// TestStepTimeout is an integration test that will verify a Step can be timed out.
+func TestStepTimeout(t *testing.T) {
+	c, namespace := setup(t)
+	t.Parallel()
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+	defer tearDown(t, c, namespace)
+
+	t.Logf("Creating Task with Step step-penguin and Step step-albatros in namespace %s", namespace)
+	taskName := "timeout"
+	task := &v1beta1.Task{
+		ObjectMeta: metav1.ObjectMeta{Name: taskName, Namespace: namespace},
+		Spec: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{
+				{Container: corev1.Container{
```

Comment:
Readability nit: I think it might be a bit easier to scan this if you change the brackets and newlines a bit, and use script mode instead of `sh -c`, to something like:

```
Steps: []v1beta1.Step{{
  Container: corev1.Container{
    Image: "busybox",
  },
  Script: "sleep 1",
  Timeout: "1ms",
}, {
  Container: corev1.Container{
    Image: "busybox",
  },
  Script: "sleep 1",
}}
```